pyquil/api/_qam.py

```diff
@@ -64,8 +70,12 @@ def get_result(self, execute_response: T) -> QAMExecutionResult:
         :param execute_response: The return value from a call to ``execute``.
         """
 
-    def run(self, executable: QuantumExecutable) -> QAMExecutionResult:
+    def run(
+        self,
+        executable: QuantumExecutable,
+        memory_map: Optional[Dict[str, Union[Sequence[int], Sequence[float]]]] = None,
```

Comment:
Could be worth defining a type alias for this? Just in case we change it in the future.

Reply:
Yeah, good idea. Added one.